pipeline-runner/R/copy-1-copy_s3_objects.R

```diff
@@ -48,23 +49,28 @@ copy_cell_sets <- function(from_experiment_id, to_experiment_id, sample_ids_map,
 }
 
 copy_source <- function(from_experiment_id, to_experiment_id, sample_ids_map, pipeline_config) {
-  replace_sample_ids <- function(key) {
-    new_key <- key
-    for (sample_id in names(sample_ids_map)) {
-      new_key <- gsub(sample_id, sample_ids_map[[sample_id]], new_key)
-    }
+  bucket <- pipeline_config$source_bucket
 
-    return(new_key)
-  }
+  for (original_sample_id in names(sample_ids_map)) {
+    copy_sample_id <- sample_ids_map[[original_sample_id]]
 
-  s3_copy_by_prefix(
-    pipeline_config$source_bucket,
-    from_experiment_id,
-    pipeline_config$source_bucket,
-    to_experiment_id,
-    pipeline_config$aws_config,
-    key_transform = replace_sample_ids
-  )
```

Comment:
Explanation for alex: these copy changes are a fix to the existing copy which wasn't working correctly (the source and cell sets weren't being copied correctly)